core/handlers.py

```diff
@@ -15,19 +21,27 @@ def handle_validation_error(validation_error: ValidationError):
     }, 400
 
 
-def handle_exception(uncaught_exception: Exception):
+def handle_exception(uncaught_exception: Exception) -> Exception | tuple[Response, int]:
     # pass through HTTP errors
     if isinstance(uncaught_exception, HTTPException):
         return uncaught_exception
 
     # handle uncaught ingest errors
     if request.path == "/ingest":
-        # TODO: save file to S3
-        current_app.logger.error("Uncaught ingest exception.", exc_info=True)
-        return {
-            "detail": "Uncaught ingest exception.",
-            "status": 500,
-            "title": "Internal Server Error",
-        }, 500
+        failure_id = uuid.uuid4()
+        s3_object_name = FAILED_FILE_S3_NAME_FORMAT.format(failure_id, datetime.now().strftime(DATETIME_ISO_8610))
```

Comment:
I forgot to mention this before, but Matt noticed there is a typo in this global name, should actually be DATETIME_ISO_8601. Could you refactor this across repo as part of this pr?